### PR TITLE
Move android release template out of build-tools.

### PR DESCRIPTION
**Pull request by mitchdenny**
This is the first in a series of PRs that I want to land where I move the release templates for Android out of the build-tools repository.

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-android-release.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  # in this case we do GPG signing in order to publish to Maven Central. At the moment signing
+  # is protected by an approval check but this may be removed in the future.
+  - stage: Signing
+    dependsOn: ${{parameters.DependsOn}}
+    jobs:
+      - deployment: SignPackage
+        environment: esrp
+        timeoutInMinutes: 20
+        pool:
+          vmImage: ubuntu-16.04
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-android-release.yml:17]:
```suggestion
          vmImage: ubuntu-18.04
```

**Review by weshaggard — APPROVED** [review]
Looks like you need to update some of the vmImages to match the rest of the repos but otherwise I assume is is a direct copy so looks good.

File: eng/pipelines/templates/stages/archetype-android-release.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            displayName: "Create release tag"
+            condition: ne(variables['Skip.TagRepository'], 'true')
+            environment: github
+            timeoutInMinutes: 5
+            dependsOn:
+              - ${{ if eq(parameters.VerifyVersions, 'true')}}:
+                - VerifyReleaseVersion
+
+            pool:
+              vmImage: vs2017-win2016
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-android-release.yml:62]: Looks like we need to probably update most of the images

Comment (mitchdenny) [reply eng/pipelines/templates/stages/archetype-android-release.yml:62]: Yeah it is all a bit stale. My goal here though is to get this stuff moved over first and then start doing a bit of a facelift.

**Pull request merged**